criu/kerndat.c

```diff
@@ -949,8 +952,10 @@ int kerndat_has_thp_disable(void)
 		return -1;
 	}
 
-	if (prctl(PR_SET_THP_DISABLE, 0, 0, 0, 0))
-		return -1;
+	if (prctl(PR_SET_THP_DISABLE, 0, 0, 0, 0)) {
+		pr_err("prctl failed.\n");	// needs review
+		goto out_unmap;	// may revert to return -1
```

Comment:
Not sure why this was previously `return -1;` without unmapping. I added `goto out_unmap;` but if there's a reason why it was initially `return -1`, I'll revert it.

Reply:
it was a bug